hack/e2e.go

```diff
@@ -211,6 +211,20 @@ func run(deploy deployer) error {
 		if err := xmlWrap("Up", deploy.Up); err != nil {
 			return fmt.Errorf("starting e2e cluster: %s", err)
 		}
+		if *dump != "" {
```

Comment:
Do we maybe want to pull this out to one level higher and run it in either the case that `-up` **or** `-test` is selected, then just eliminate the `kubectl version` check for `-test`? i.e. move it to about line 215 in the original and just say:
```
if *up || *test {
  // get nodes. If there's something wrong, we need to get out of dodge after this.
  // if *dump is set, send the output to a file
}
```